core/src/main/java/org/apache/iceberg/UpdateRequirement.java

```diff
@@ -114,6 +117,15 @@ public void validate(TableMetadata base) {
             lastAssignedFieldId, base.lastColumnId());
       }
     }
+
+    @Override
+    public void validate(ViewMetadata base) {
+      if (base != null && base.schema().highestFieldId() != lastAssignedFieldId) {
```

Comment:
This isn't correct, but it probably doesn't matter. Consider a case where I add a new field (id=3) and then roll back to a previous schema where the highest field ID is 2. That's why we store the last assigned field ID separately in the table spec. However, I don't think that we would ever use this validation for views. We don't store data files that embed schema field IDs, so we don't have any need for them to be consistent across versions or schemas. I think this should just throw an `UnsupportedOperationException ` because we should not generated the validation.

(We could also assign field IDs like we do in the table schema, but I'd probably skip that complexity)

Reply:
yes I agree this isn't correct, that's why I added a TODO to double-check this during review. Thanks for clarifying